zerver/views/thumbnail.py

```diff
@@ -24,11 +36,14 @@ def validate_thumbnail_request(user_profile: UserProfile, path: str) -> Optional
 @has_request_variables
 def backend_serve_thumbnail(
     request: HttpRequest,
-    user_profile: UserProfile,
+    maybe_user_profile: Union[UserProfile, AnonymousUser],
     url: str = REQ(),
     size_requested: str = REQ("size"),
 ) -> HttpResponse:
-    if not validate_thumbnail_request(user_profile, url):
+    if not maybe_user_profile.is_authenticated:
+        maybe_user_profile.realm = get_valid_realm_from_request(request)
```

Comment:
Yeah, it is a common patter we use for different methods too. So, it doesn't make sense to modify each of them.

Reply:
I'm not sure writing to `AnonymousUser` is safe -- does Django give you a new one for every request?

Reply:
In any case I think it'd be cleaner to just pass `realm` as a separate parameter passed to `validate_attachment_request`.  I think shifting to do this would be a good preparatory refactor.

Reply:
Can't do it as a preparatory refactor due to coverage requirements.

Reply:
I don't see any other uses of this pattern currently.  Did we decide this was safe?